### PR TITLE
Refactor action type

### PR DESCRIPTION
**Pull request by cniska**

### CODE REVIEW

**Comment by cniska** [comment]
@AriaMinaei can you check that this looks fine and merge it. Thanks!

File: src/hello/reducer.js [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 import {combineReducers} from 'redux'
 
-import type {ReplyAction} from './types'
+import type {Action} from '../types'
 
-export const messageReducer = (state: string = '', {type, payload: message}: ReplyAction) => {
+export const messageReducer = (state: string = '', {type, payload: message}: $Subtype<Action>) => {
```
Comment (AriaMinaei) [inline src/hello/reducer.js]: We are losing type data here. I like the fact that by not defining action type constants and individual action types, we are typing less, but the downside then would be that we lose type info.

**Review by AriaMinaei — DISMISSED** [review]

File: src/hello/reducer.js (hunk 1 above)
Comment (cniska) [reply src/hello/reducer.js]: The problem with specific action types in the `payload` property. What would be the best way to handle that, i.e. how should we define the `Action` type for it to work correctly?

File: src/hello/selectors.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
 import type {Selector} from '../../types'
 
-export const getMessage: Selector<string, void> = (state) =>
+export const getMessage: Selector<string> = (state) =>
```
Comment (AriaMinaei) [inline src/hello/selectors.js:3]: Does this pass flow? Both args are required I think

Comment (cniska) [reply src/hello/selectors.js:3]: It does pass.

**Pull request merged**